pootle/apps/pootle_app/management/commands/contributors.py

```diff
@@ -32,107 +31,77 @@ def add_arguments(self, parser):
-
-        submission = submissions_qs.last()
-
-        if submission is None:
-            return 0
-
-        return submission.unit.revision
-
-    def handle_all(self, **options):
-        system_user = User.objects.get_system_user()
-        units = Unit.objects.exclude(submitted_by=system_user) \
-                            .exclude(submitted_by=None)
-
-        if options["only_emails"]:
-            nobody_user = User.objects.get_nobody_user()
-            units = units.exclude(submitted_by=nobody_user)
-
-        if options["since"]:
-            units = units.filter(
-                revision__gte=self._get_revision_from_since(options["since"]),
-            )
-
-        if self.projects:
-            units = units.filter(
-                store__translation_project__project__code__in=self.projects,
-            )
-
-        if self.languages:
-            units = units.filter(
-                store__translation_project__language__code__in=self.languages,
-            )
-
-        contribs = Counter()
-        for v in units.values("submitted_by"):
-            contribs.update((v["submitted_by"], ))
-
-        self.list_contributions(contribs, options["sort_by"],
-                                options["only_emails"])
-
-    def list_contributions(self, contribs, sort_by, only_emails):
-        if sort_by == "name":
-            contributions = contribs.items()
-        else:
-            contributions = contribs.most_common()
-
-        out = []
-        for id, count in contributions:
-            user = User.objects.get(id=id)
-            name = user.display_name
-            if only_emails:
-                name = name.replace(",", "")
-            if user.email:
-                name += " <%s>" % (user.email)
-            elif only_emails:
-                continue
-            if not only_emails:
-                name = "%s (%i contributions)" % (name, count)
-            out.append(name)
-
-        if sort_by == "name":
-            # Sort users alphabetically
-            out = sorted(out)
-
-        for line in out:
-            self.stdout.write(line)
+    def get_aware_datetime(self, dt_string):
+        """Return an aware datetime parsed from a datetime string.
+
+        Date string can be any format parsable by dateutil.parser.parse
+        """
+        if not dt_string:
+            return None
+        try:
+            return make_aware(parse_datetime(dt_string))
+        except ValueError:
+            raise CommandError("The provided datetime/date string is not "
+                               "valid: %s" % dt_string)
+
+    @property
+    def contributors(self):
+        return contributors.get()
+
+    def contrib_kwargs(self, **options):
+        kwargs = {
+            k: v
+            for k, v
+            in options.items()
+            if k in ["projects", "languages", "include_anon",
+                     "since", "until", "sort_by"]}
+        kwargs["since"] = self.get_aware_datetime(kwargs["since"])
```

Comment:
You could probably use [type=get_aware_datetime](https://docs.python.org/2/library/argparse.html#type) to achieve the same